### PR TITLE
Frontend/feature/#1277 Driver Profile revamp

### PR DESCRIPTION
**Pull request by aryan-13**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description

Driver profile revamp
Figma - [Link](https://www.figma.com/file/4zT4QhImxqphGf56DpAjcV/Beckn-Release-Version?type=design&node-id=61198-14660&mode=design&t=eTJbi9wS8LgnJqA1-4)
Output - [Link](https://drive.google.com/file/d/1VHFw6vsMkAKIdkovRz3BqwoWGkI2mDTk/view?usp=sharing)
### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/ScreenData.purs [hunk 1]
```diff
@@ -172,7 +172,7 @@ initData = {
     , isBanner : true
     , callSupportPopUp : false
     , isMockLocation: false
-    , isSpecialZone : false
+    , isSpecialZone : true
```
Comment (witcher-shailesh) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/ScreenData.purs:175]: why default value of SpecialZone is `true` ?

File: Frontend/ui-driver/src/App.purs [hunk 2]
```diff
@@ -198,6 +198,12 @@ data DRIVER_PROFILE_SCREEN_OUTPUT = DRIVER_DETAILS_SCREEN
                                     | NOTIFICATIONS_SCREEN
                                     | GO_TO_REFERRAL_SCREEN_FROM_DRIVER_PROFILE_SCREEN
                                     | GO_TO_BOOKING_OPTIONS_SCREEN DriverProfileScreenState
+                                    | VERIFY_OTP1 DriverProfileScreenState
```
Comment (witcher-shailesh) [inline Frontend/ui-driver/src/App.purs:201]: Try not to use bad nomenclature

File: Frontend/ui-driver/src/Screens/DriverProfileScreen/ScreenData.purs [hunk 3]
```diff
@@ -66,19 +89,47 @@ type Listtype =
 optionList :: String -> Array Listtype
 optionList dummy =
     [
-      {menuOptions: DRIVER_PRESONAL_DETAILS , icon:"ny_ic_profile,https://assets.juspay.in/nammayatri/images/driver/ny_ic_profile.png"},
-      {menuOptions: DRIVER_VEHICLE_DETAILS , icon:"ny_ic_car_profile,https://assets.juspay.in/nammayatri/images/driver/ny_ic_car_profile.png"}
+      -- {menuOptions: DRIVER_PRESONAL_DETAILS , icon:"ny_ic_profile,https://assets.juspay.in/nammayatri/images/driver/ny_ic_profile.png"},
```
Comment (witcher-shailesh) [inline Frontend/ui-driver/src/Screens/DriverProfileScreen/ScreenData.purs:92]: If this isn't required, can you remove it? If req in future, then add proper message

File: Frontend/android-native/app/src/driver/jatriSaathiDriver/res/drawable/ic_settings.png
Comment (Vignesh-772) [inline Frontend/android-native/app/src/driver/jatriSaathiDriver/res/drawable/ic_settings.png:1]: reduce the image size

File: Frontend/ui-driver/src/Screens/DriverProfileScreen/View.purs [hunk 4]
```diff
@@ -72,44 +91,597 @@ screen initialState =
...
+          ][ PopUpModal.view (push <<<PopUpModalAction) (logoutPopUp state) ]
+        , if state.props.showLiveDashboard then showLiveStatsDashboard push state else dummyTextView 
+        -- , if state.props.updateDetails then updateDetailsView state push else dummyTextView
+        , if state.props.showGenderView || state.props.alternateNumberView then driverNumberGenderView state push else dummyTextView
+        , if state.props.removeAlternateNumber then PopUpModal.view (push <<<  RemoveAlternateNumberAC) (removeAlternateNumberConfig state ) else dummyTextView
+        , if state.props.enterOtpModal then enterOtpModal push state else dummyTextView
+        , if state.props.updateLanguages then updateLanguageView state push else dummyTextView]
+
+-- updateDetailsView :: forall w. ST.DriverProfileScreenState -> (Action -> Effect Unit )-> PrestoDOM (Effect Unit) w
+-- updateDetailsView state push =
```
Comment (witcher-shailesh) [inline Frontend/ui-driver/src/Screens/DriverProfileScreen/View.purs:123]: Remove code, which is commented and not required

File: Frontend/android-native/app/src/driver/nammaYatriPartner/res/drawable/ny_ic_navigator_badge.png
Comment (Vignesh-772) [inline Frontend/android-native/app/src/driver/nammaYatriPartner/res/drawable/ny_ic_navigator_badge.png:1]: reduce all the image sizes

File: Frontend/ui-driver/src/Flow.purs [hunk 5]
```diff
@@ -679,6 +679,95 @@ driverProfileFlow = do
       modifyScreenState $ BookingOptionsScreenType (\bookingOptions -> bookingOptions{data{vehicleType = state.data.driverVehicleType, vehicleNumber = state.data.vehicleRegNumber, vehicleName = state.data.vehicleModelName, vehicleCapacity = state.data.capacity, downgradeOptions = ((downgradeOptionsConfig state.data.vehicleSelected) <$> state.data.downgradeOptions)}})
       bookingOptionsFlow
 
+    DRIVER_ALTERNATE_CALL_API1 updatedState -> do
```
Comment (Vignesh-772) [inline Frontend/ui-driver/src/Flow.purs:682]: `DRIVER_ALTERNATE_CALL_API1` replace this with something like  `DRIVER_ALTERNATE_CALL_API_ONE` or something meaningful

File: Frontend/ui-driver/src/Screens/DriverProfileScreen/ScreenData.purs [hunk 6]
```diff
@@ -66,19 +89,47 @@ type Listtype =
...
+      -- {menuOptions: DRIVER_VEHICLE_DETAILS , icon:"ny_ic_car_profile,https://assets.juspay.in/nammayatri/images/driver/ny_ic_car_profile.png"}
```
Comment (Vignesh-772) [inline Frontend/ui-driver/src/Screens/DriverProfileScreen/ScreenData.purs:93]: remove this if it not required

File: Frontend/ui-driver/src/Screens/DriverProfileScreen/View.purs [hunk 7]
```diff
@@ -72,44 +91,597 @@ screen initialState =
...
+                , background Color.blue600
+                , padding $ Padding 12 10 12 10
+                , gravity CENTER_VERTICAL
+                -- , margin $ MarginHorizontal 16 10
+                ][  textView
+                    [ text "243"
+                    , width WRAP_CONTENT
+                    , height WRAP_CONTENT
+                    , textSize FontSize.a_14
+                    , fontStyle $ FontStyle.bold LanguageStyle
```
Comment (Vignesh-772) [inline Frontend/ui-driver/src/Screens/DriverProfileScreen/View.purs:395]: Append the fontStyles don't hard code

File: Frontend/ui-driver/src/Screens/DriverProfileScreen/View.purs [hunk 8]
```diff
@@ -230,45 +780,351 @@ profileOptionsLayout state push =
...
+          ]
+      ]
+
   ]
 
 
---------------------------------------------------------------- horizontalLineView and dummyTextView ----------------------------
+------------------------------------------ ANIMATION -----------------------------------------------------
+
+addAnimation state = PrestoAnim.animationSet [ Anim.fadeOut (state.props.screenType == ST.AUTO_DETAILS), Anim.fadeOut (state.props.screenType == ST.DRIVER_DETAILS), Anim.fadeIn (state.props.screenType == ST.AUTO_DETAILS), Anim.fadeOut (state.props.screenType == ST.DRIVER_DETAILS), Anim.fadeIn (state.props.screenType == ST.DRIVER_DETAILS)]
```
Comment (Vignesh-772) [inline Frontend/ui-driver/src/Screens/DriverProfileScreen/View.purs:1060]: Add proper Function Signature

**Review by Vignesh-772 — CHANGES_REQUESTED** [review]

File: Frontend/ui-driver/src/Flow.purs (hunk 5 above)
Comment (witcher-shailesh) [reply Frontend/ui-driver/src/Flow.purs:682]: DRIVER_ALTERNATE_CALL_API1 and DRIVER_ALTERNATE_CALL_API_ONE is same @Vignesh-772 where `_API1` looks better than _ONE

**Review by PraveenGongada — CHANGES_REQUESTED** [review]

File: Frontend/ui-driver/src/Screens/DriverProfileScreen/View.purs (hunk 4 above)
Comment (aryan-13) [reply Frontend/ui-driver/src/Screens/DriverProfileScreen/View.purs:123]: This is required

**Review by Vignesh-772 — APPROVED** [review]

**Review by PraveenGongada — APPROVED** [review]

File: Frontend/android-native/app/src/driver/nammaYatriPartner/res/drawable/ny_ic_auto_side_view.png
Comment (Rohit4997) [inline Frontend/android-native/app/src/driver/nammaYatriPartner/res/drawable/ny_ic_auto_side_view.png:1]: Can we reduce the image size here?

File: Frontend/android-native/app/src/driver/nammaYatriPartner/res/drawable/ny_ic_clean_auto_badge.png
Comment (Rohit4997) [inline Frontend/android-native/app/src/driver/nammaYatriPartner/res/drawable/ny_ic_clean_auto_badge.png:1]: Can we reduce the image size here?

File: Frontend/android-native/app/src/driver/nammaYatriPartner/res/drawable/ny_ic_expert_driver.png
Comment (Rohit4997) [inline Frontend/android-native/app/src/driver/nammaYatriPartner/res/drawable/ny_ic_expert_driver.png:1]: Can we reduce the image size here?

File: Frontend/android-native/app/src/driver/nammaYatriPartner/res/drawable/ny_ic_five_star_badge.png
Comment (Rohit4997) [inline Frontend/android-native/app/src/driver/nammaYatriPartner/res/drawable/ny_ic_five_star_badge.png:1]: Can we reduce the image size here?

File: Frontend/android-native/app/src/driver/nammaYatriPartner/res/drawable/ny_ic_navigator_badge.png
Comment (Rohit4997) [inline Frontend/android-native/app/src/driver/nammaYatriPartner/res/drawable/ny_ic_navigator_badge.png:1]: Can we reduce the image size here?

File: Frontend/android-native/app/src/driver/nammaYatriPartner/res/drawable/ny_ic_ontime_badge.png
Comment (Rohit4997) [inline Frontend/android-native/app/src/driver/nammaYatriPartner/res/drawable/ny_ic_ontime_badge.png:1]: Can we reduce the image size here?

File: Frontend/android-native/app/src/driver/nammaYatriPartner/res/drawable/ny_ic_polite_driver_badge.png
Comment (Rohit4997) [inline Frontend/android-native/app/src/driver/nammaYatriPartner/res/drawable/ny_ic_polite_driver_badge.png:1]: Can we reduce the image size here?

File: Frontend/android-native/app/src/driver/nammaYatriPartner/res/drawable/ny_ic_safe_ride.png
Comment (Rohit4997) [inline Frontend/android-native/app/src/driver/nammaYatriPartner/res/drawable/ny_ic_safe_ride.png:1]: Can we reduce the image size here?

File: Frontend/ui-driver/src/Screens/DriverProfileScreen/View.purs [hunk 9]
```diff
@@ -72,44 +91,588 @@ screen initialState =
...
+        , textSize FontSize.a_14
+        , color Color.black800
+        , height WRAP_CONTENT
+        , fontStyle $ FontStyle.medium LanguageStyle
+        ]
     ]
 
 
+------------------------------------------------- BADGE DATA -----------------------------------------------------------------
+getBadgeData :: ST.DriverProfileScreenState -> Array {badgeImage :: String, primaryText :: String, subText :: String}
```
Comment (Rohit4997) [inline Frontend/ui-driver/src/Screens/DriverProfileScreen/View.purs:468]: Can we move this to componentConfig file?

File: Frontend/ui-driver/src/Screens/DriverProfileScreen/View.purs [hunk 10]
```diff
@@ -230,45 +771,349 @@ profileOptionsLayout state push =
...
+      ]
+
   ]
 
 
---------------------------------------------------------------- horizontalLineView and dummyTextView ----------------------------
+------------------------------------------ ANIMATION -----------------------------------------------------
+addAnimation state = PrestoAnim.animationSet [ Anim.fadeOut (state.props.screenType == ST.AUTO_DETAILS), Anim.fadeOut (state.props.screenType == ST.DRIVER_DETAILS), Anim.fadeIn (state.props.screenType == ST.AUTO_DETAILS), Anim.fadeOut (state.props.screenType == ST.DRIVER_DETAILS), Anim.fadeIn (state.props.screenType == ST.DRIVER_DETAILS)] 
+
+scaleUpConfig :: Boolean -> AnimConfig.AnimConfig
```
Comment (Rohit4997) [inline Frontend/ui-driver/src/Screens/DriverProfileScreen/View.purs:1051]: Can we move this to componentConfig file?

File: Frontend/ui-driver/src/Screens/DriverProfileScreen/View.purs [hunk 11]
```diff
@@ -230,45 +771,349 @@ profileOptionsLayout state push =
...
+        , toScaleX = 1.0 
+        , toScaleY = 1.0 
+        , fromScaleY = 0.5
+        , fromScaleX = 0.5
+        , ifAnim = ifAnim
+        }
+  in 
+    animConfig'
+
+scaleDownConfig :: Boolean -> AnimConfig.AnimConfig
```
Comment (Rohit4997) [inline Frontend/ui-driver/src/Screens/DriverProfileScreen/View.purs:1067]: Can we move this to componentConfig file?

File: Frontend/ui-driver/src/Screens/DriverProfileScreen/View.purs [hunk 12]
```diff
@@ -230,45 +771,349 @@ profileOptionsLayout state push =
...
+        , fromScaleY = 1.0
+        , fromScaleX = 1.0
+        , ifAnim = ifAnim
+        }
+  in 
+    animConfig'
+
+---------------------------------------------- Data ARRAY -----------------------------------------------------
+
+driverDetailsArray :: forall w. ST.DriverProfileScreenState -> Array {key :: String , value :: Maybe String , action :: Action, isEditable :: Boolean}
```
Comment (Rohit4997) [inline Frontend/ui-driver/src/Screens/DriverProfileScreen/View.purs:1085]: Can we move this to componentConfig file?

File: Frontend/android-native/app/src/driver/nammaYatriPartner/res/drawable/ny_ic_clean_auto_badge.png
Comment (aryan-13) [reply Frontend/android-native/app/src/driver/nammaYatriPartner/res/drawable/ny_ic_clean_auto_badge.png:1]: Reduced already for all images

**Pull request merged**